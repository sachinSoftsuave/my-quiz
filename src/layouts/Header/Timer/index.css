.timer{
     cursor: pointer;
     text-decoration: none;
     font-weight: 900;
     font-family: Verdana, Geneva, Tahoma, sans-serif;
     font-size: 15px;
     color: white;
 }