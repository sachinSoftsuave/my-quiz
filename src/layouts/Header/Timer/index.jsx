import React, {useState, useEffect, useContext} from "react";
import { useNavigate } from "react-router-dom";
import { QuizContext } from "../../../hooks/context";
import "./index.css";

function Timer() {

  const [minutes, setMinutes] = useState(50);
  const [isActive, setIsActive] = useState(false);
  const {isQuizStarted} = useContext(QuizContext);
  let navigate = useNavigate();
  console.log(isQuizStarted,"isQuizStarted");

  useEffect(() => {
    if(isQuizStarted)
    {
      setMinutes(50)
      startTimer()
    }
  },[isQuizStarted])

  function startTimer() {
    setIsActive(!isActive);
  }

  useEffect(() => {
    let interval = null;
    if (isActive && minutes > 0) {
      interval = setInterval(() => {
        setMinutes(minutes => minutes - 1);
      }, 1000);
    } else if(minutes <=0) {
      clearInterval(interval);
      setIsActive(false);
      navigate('/results')
    }
    return () => clearInterval(interval);
  }, [isActive, minutes]);

  return (
    <div>
      <p className="timer"> Time:{minutes} </p>
    </div>
  );
}

export default Timer;
