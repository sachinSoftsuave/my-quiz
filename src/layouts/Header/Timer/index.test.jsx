import React from "react";
import { mount } from 'enzyme';
import "jsdom-global/register";
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Timer from "./index.jsx";
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useContext: () => ({
  isQuizStarted: false
  }),
}));


const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

describe("Timer component", () => {
  let wrapper;
    beforeEach(() => {
         wrapper = mount(<Timer/>)
    })
  
  it('it render Timer component', () =>{
        expect(wrapper.exists()).toBe(true);
  })

  it('render the title of timer', () => {
      expect(wrapper.find('.timer').text()).toContain(' Time: ')
  })

  it('renders the snapshot of timer component', () =>{
        expect(toJson(wrapper)).toMatchSnapshot();
  })

});
