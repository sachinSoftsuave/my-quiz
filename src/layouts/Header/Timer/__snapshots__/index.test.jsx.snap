// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer component renders the snapshot of timer component 1`] = `
<Timer>
  <div>
    <p
      className="timer"
    >
       Time:
       
    </p>
  </div>
</Timer>
`;
